internal/cli/atlas/quickstart/quick_start.go

```diff
@@ -0,0 +1,353 @@
+		DiskSizeGB:          &diskSizeGB,
+		Name:                opts.ClusterName,
+	}
+}
+
+func (opts *Opts) newReplicationSpec() atlas.ReplicationSpec {
+	var (
+		readOnlyNodes int64 = 0
+		priority      int64 = 7
+		shards        int64 = shards
+		members       int64 = members
+	)
+	replicationSpec := atlas.ReplicationSpec{
+		NumShards: &shards,
+		ZoneName:  zoneName,
+		RegionsConfig: map[string]atlas.RegionsConfig{
+			opts.Region: {
+				ReadOnlyNodes:  &readOnlyNodes,
+				ElectableNodes: &members,
+				Priority:       &priority,
+			},
+		},
+	}
+	return replicationSpec
+}
+
+func (opts *Opts) newProviderSettings() *atlas.ProviderSettings {
+	return &atlas.ProviderSettings{
+		InstanceSizeName: tier,
+		ProviderName:     opts.Provider,
+		RegionName:       opts.Region,
+	}
+}
+
+// askRequiredFlags allows the user to set required flags by using interactive prompts
+func (opts *Opts) askRequiredFlags() error {
+	qs := opts.newDBUserQuestions()
+	qs = append(qs, opts.newAccessListQuestion())
+	qs = append(qs, opts.newClusterQuestions()...)
+
+	if err := survey.Ask(qs, opts); err != nil {
+		return err
+	}
+
+	// we call survey.Ask two times because the region question needs opts.Provider to be populated
+	if err := survey.Ask([]*survey.Question{opts.newRegionQuestions()}, opts); err != nil {
+		return err
+	}
+
+	if opts.DBUserPassword == "" {
+		pwd, err := randgen.GenerateRandomBase64String(passwordLength)
+		if err != nil {
+			return err
+		}
+		opts.DBUserPassword = pwd
+	}
+
+	return nil
+}
+
+func (opts *Opts) newDBUserQuestions() []*survey.Question {
+	var qs []*survey.Question
+	if opts.DBUsername == "" {
+		q := []*survey.Question{
+			{
+				Name: "dbUsername",
+				Prompt: &survey.Input{
+					Message: "Insert the Username for authenticating to MongoDB",
+					Help:    usage.DBUsername,
+					Default: "quickStartUser",
+				},
+			},
+		}
+
+		qs = append(qs, q...)
+	}
+
+	if opts.DBUserPassword == "" {
+		q := []*survey.Question{
+			{
+				Name: "dbUserPassword",
+				Prompt: &survey.Password{
+					Message: "Insert the Password for authenticating to MongoDB [Press Enter to use an autogenerated password]",
+					Help:    usage.Password,
+				},
+			},
+		}
+
+		qs = append(qs, q...)
+	}
+
+	return qs
+}
+
+func (opts *Opts) newAccessListQuestion() *survey.Question {
+	if opts.IPAddress != "" {
+		return nil
+	}
+	publicIP, _ := net.NewIPAddress()
+	return &survey.Question{
+
+		Name: "ipAddress",
+		Prompt: &survey.Input{
+			Message: "Insert the IP entry to add to the Access List",
+			Help:    usage.AccessListIPEntry,
+			Default: publicIP,
+		},
+	}
+}
+
+func (opts *Opts) newClusterQuestions() []*survey.Question {
+	var qs []*survey.Question
+
+	if opts.ClusterName == "" {
+		q := []*survey.Question{
+			{
+				Name: "clusterName",
+				Prompt: &survey.Input{
+					Message: "Insert the cluster name",
+					Help:    usage.Provider,
+					Default: clusterName,
+				},
+			},
+		}
+		qs = append(qs, q...)
```

Comment:
I have not fully reviewed this but FYI you do `q...` because is a slice, you can simplify this to be 
```suggestion
		q := &survey.Question{
			Name: "clusterName",
			Prompt: &survey.Input{
				Message: "Insert the cluster name",
				Help:    usage.Provider,
				Default: clusterName,
			},
		}
		qs = append(qs, q)
```